src/deploy/builder.ts

```diff
@@ -1,58 +1,27 @@
 import {
   BuilderContext,
   BuilderOutput,
-  createBuilder
+  createBuilder,
 } from '@angular-devkit/architect';
-import { asWindowsPath, experimental, normalize } from '@angular-devkit/core';
-import { NodeJsSyncHost } from '@angular-devkit/core/node';
-import os from 'os';
-import * as path from 'path';
 
 import * as engine from '../engine/engine';
 import deploy from './actions';
 import { Schema } from './schema';
 
 // Call the createBuilder() function to create a builder. This mirrors
 // createJobHandler() but add typings specific to Architect Builders.
-export default createBuilder<any>(
+export default createBuilder(
   async (options: Schema, context: BuilderContext): Promise<BuilderOutput> => {
-    try {
-      // The project root is added to a BuilderContext.
-      const root = normalize(context.workspaceRoot);
-      const workspace = new experimental.workspace.Workspace(
-        root,
-        new NodeJsSyncHost()
-      );
-      await workspace
-        .loadWorkspaceFromHost(normalize('angular.json'))
-        .toPromise();
-
-      if (!context.target) {
-        throw new Error('Cannot deploy the application without a target');
-      }
-
-      const targets = workspace.getProjectTargets(context.target.project);
-
-      if (
-        !targets ||
-        !targets.build ||
-        !targets.build.options ||
-        !targets.build.options.outputPath
-      ) {
-        throw new Error('Cannot find the project output directory');
-      }
+    if (!context.target) {
+      throw new Error('Cannot deploy the application without a target');
+    }
 
-      const isWin = os.platform() === 'win32';
-      const workspaceRoot = !isWin
-        ? workspace.root
-        : asWindowsPath(workspace.root);
+    const buildTarget = {
+      name: `${context.target.project}:build:production`,
```

Comment:
i will extend this to

```ts
const buildTarget = { name: options.buildTarget || `${context.target.project}:build:production` };
```

as seen here: https://github.com/angular/angularfire/commit/7e1918a24653a21e46e8ce152ee1c23142684b62#diff-77ce940ab1bf28b9fd490f028b74d4a439968ace1f095f72216d64f51bed7c59R25